app/workers/new_page_worker.rb

```diff
@@ -0,0 +1,45 @@
+class NewPageWorker
+  include Sidekiq::Worker
+  attr_reader :page, :doc_html
+
+  def perform(page_id)
+    @page = Page.find(page_id)
+    @doc_html = download_content
+    process
+  rescue SocketError, Gastly::FetchError, OpenURI::HTTPError, Net::OpenTimeout
+    page.update_attributes(status: Page::ERROR_STATUS)
+  end
+
+  private
+
+  def process
+    page.update_attributes(status: Page::IN_PROGRESS_STATUS)
+    parse_html_content
+    make_screenshot
+    page.update_attributes(status: Page::PROCESSED_STATUS)
+  end
+
+  def download_screenshot(file)
+    File.new(Gastly.capture(page.source_url, file.path))
+  end
+
+  def make_screenshot
+    file = Tempfile.new
+    page.update_attributes(screenshot: download_screenshot(file))
+    file.close
+    file.unlink
```

Comment:
[`Tempfile#close!`](http://ruby-doc.org/stdlib-1.9.3/libdoc/tempfile/rdoc/Tempfile.html#method-i-close-21) instead of `#close` + `#unlink`

Reply:
+